src/aind_metadata_upgrader/data_description_upgrade.py

```diff
@@ -143,7 +143,10 @@ def get_creation_time(self, **kwargs):
         old_name = self._get_or_default(self.old_model, "name", kwargs)
         if creation_date is not None:
             creation_date = datetime.strptime(creation_date, "%Y-%m-%d").date()
-            creation_time = datetime.strptime(creation_time, "%H:%M:%S").time()
+            if '.' in creation_time:
```

Comment:
Instead of trying to detect millis with the period, could you always try to use the  `%H:%M:%S.%f` format and just zero out the millis?

Reply:
this is weird. I thought I added this comment, but I don't see it:
```python
# This works for six decimal places
creation_date = "2020-10-10"
creation_time = "16:01:12.123456"
datetime.fromisoformat(f"{creation_date}T{creation_time}")

# It also works if only seconds are provided
creation_date = "2020-10-10"
creation_time = "16:01:12"
datetime.fromisoformat(f"{creation_date}T{creation_time}")

# For less than six digits, it's safer to use regex pattern matching rather than checking if the string contains a period
```

Reply:
I like Jon's